public/js/surveyBuilderForm.js

```diff
@@ -0,0 +1,190 @@
+                    options.each(function(key,option)
+                    {
+                        response += '<div><input type="' + type + '" disabled value="' + option.value + '"> ' + option.value + '</div>'
+                    })
+                }
+            }
+
+            response += '</div>'
+
+            var newQuestion =   '<div class="new-question ui-state-default">' +
+                            '<h5>' + question + '</h5>' +
+                            response +
+                            '<input type="submit" class="remove-question btn btn-sm" value="Remove">' +
+                        '</div>'
+            
+            var $newQuestion = $(newQuestion).data('required', $required.is(':checked'))
+
+            $questionContainer.append($newQuestion)
+
+            $questionContainer.sortable(
+                {
+                    placeholder: "ui-state-highlight"
+                }
+            )
+
+            $('.remove-question').click(function()
+            {
+                $(this).parent('.new-question').remove()
+            })
+
+            //resetting form
+            $question.val('')
+            $type.val('text-input')
+            $required.prop('checked', false)
+            $typeOptions.remove()
+            addQuestionButton.prop("disabled",true)
+        }
+    )
+})
+
+
+//_________________________________FUNCTION DEFINITIONS__________________________________________
+
+
+/**
+ * Adds the facility to add and remove options for multiple choice questions. Initially this consists of a text
+ * field and an add option (+) button.
+ *
+ * @param $container OBJECT the created DOM objects are appended to the contents of this html element
+ */
+function addOptionsCreator($container)
+{
+    $container.append(
+        '<div id="question-options">' +
+        '<label for="option-text">Options:</label>' +
+        '<div id="new-option-container" class="input-group col-xs-2">' +
+        '<input type="text" id="option-text">' +
+        '<button class="btn input-group-addon" id="add-option">+</button>' +
+        '</div>' +
+        '</div>'
+    )
+
+    createAddOptionHandler()
+}
+
+
+/**
+ * Creates a handler for the click event on the add option button. The handler adds the option (providing it's
+ * non-empty) to the list of options for that question.
+ * Then validates the form.
+ */
+function createAddOptionHandler()
+{
+    $('#add-option').click(function()
+    {
+        if($('#option-text').val() !== '') // disables an empty string being added as an option
+        {
+            createOption($('#new-option-container'))
+        }
+
+        validateNewQuestion($('#question').val(), $('#add-question'), $('#input-selector').val())
+    })
+}
+
+
+/**
+ * Creates a disabled field containing the text passed in, with a remove (-) button .
+ *
+ * @param $optionContainer OBJECT the div containing the text input and add button for creating new options
+ */
+function  createOption($optionContainer)
```

Comment:
It looks weird when the function description is only on one line, but this style of having a blank line between the description and parameter list is consistent across my DocBlocks and seems (after a little googling) to be a style adopted by others.